src/Graphics/OpenGLContext/GLSL/glsl_CombinerProgramBuilder.cpp

```diff
@@ -1016,15 +1016,30 @@ class ShaderFragmentHeaderReadTex : public ShaderPart
 					shaderPart +=
 						"#define TEX_OFFSET(off, tex, texCoord) texture(tex, texCoord - (off)/texSize)									\n"
 						"#define TEX_FILTER(name, tex, texCoord)												\\\n"
-						"  {																					\\\n"
+						"{																						\\\n"
 						"  mediump vec2 texSize = vec2(textureSize(tex,0));										\\\n"
 						"  mediump vec2 offset = fract(texCoord*texSize - vec2(0.5));							\\\n"
 						"  offset -= step(1.0, offset.x + offset.y);											\\\n"
 						"  lowp vec4 c0 = TEX_OFFSET(offset, tex, texCoord);									\\\n"
 						"  lowp vec4 c1 = TEX_OFFSET(vec2(offset.x - sign(offset.x), offset.y), tex, texCoord);	\\\n"
 						"  lowp vec4 c2 = TEX_OFFSET(vec2(offset.x, offset.y - sign(offset.y)), tex, texCoord);	\\\n"
+						"																						\\\n"
+						"  bool multiplyAlpha = uEnableAlphaTest == 1 || uCvgXAlpha == 1;						\\\n"
+						"  if(multiplyAlpha){																	\\\n" // Calculate premultiplied color values
+						"    c0.rgb = c0.rgb * vec3(c0.a);														\\\n"
```

Comment:
It is enough to just 
c0.rgb *= c0.a;
without converting alpha to vector. It is allowed to multiply vector by variable.
The same below.